nc/source/engine/Engine.cpp

```diff
@@ -109,6 +123,11 @@ Transform * nc::NcGetMainCameraTransform()
     return g_EngineData.mainCameraTransform;
 }
 
+void nc::ui::NcRegisterUI(IUI* ui)
+{
+    g_EngineSystems.ui->BindProjectUI(ui);
+}
+
 EntityHandle nc::NcCreateEntity()
```

Comment:
This .cpp is getting too long, and it's getting difficult to understand what all of its concerns are.
I recommend splitting it up into Entity methods in one class, Component methods in another, and the rest of this classs as a third.